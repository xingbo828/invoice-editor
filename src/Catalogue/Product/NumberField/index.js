import React from 'react';
import {
  InputContainer,
  Quantity,
  InputWrapper,
  PlusNumberHandler,
  MinusNumberHandler,
  Icon,
  Unit
} from './styles'

const NumberField = ({ onChange, placeholder, unit, value } ) => {

  const handleClick = (interval) => (e) => {
    e.preventDefault();
    let v = value + interval >= 0 ? value + interval : value
    onChange(v)
  }
  return (
    <InputContainer>
      <InputWrapper>
        <MinusNumberHandler available={value > 0} onClick={handleClick(-1)} data-test="minusHandler">
          <Icon className="fa fa-minus" aria-hidden="true"></Icon>
        </MinusNumberHandler>
        <Quantity placeholder={placeholder}>{value}</Quantity>
        <PlusNumberHandler available onClick={handleClick(1)} data-test="plusHandler">
          <Icon className="fa fa-plus" aria-hidden="true"></Icon>
        </PlusNumberHandler>
      </InputWrapper>
      <Unit>{unit}</Unit>
    </InputContainer>
  )
}

NumberField.defaultProps = {
  placeholder: '',
  label: '',
  value: 0,
};

export default NumberField;