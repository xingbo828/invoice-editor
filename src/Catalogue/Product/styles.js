import Styled from 'styled-components'

export const Container = Styled.li`
  width: 100%;
  text-align: center;
  margin-bottom: 1rem;
  display: inline-block;
  padding: 2rem;
  box-shadow: ${props=>props.theme.boxShadow.large};
  border-radius: 4px;
  ${props=>props.theme.media.greaterThan('sm')`
    flex: 1 0 30%;
  `}
`
export const Title = Styled.div`
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ${props=>props.theme.colors.textDark};
`

export const Price = Styled.div`
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ${props=>props.theme.colors.textDark};
`

export const Image = Styled.div`
  margin: 0 auto 1.5rem;
  width: 100%;
  height: 100px;
  background: url('${props=>props.src}') center center no-repeat;
  background-size: contain;
`

