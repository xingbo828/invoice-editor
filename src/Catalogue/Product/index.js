import { compose, withHandlers } from 'recompose'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { updateProduct } from '../../cartAction'
import { getProductQuantity } from '../../cartReducer'

import Product from './Product'

const mapDispatchToProps = dispatch => bindActionCreators({
  updateProduct
}, dispatch)

const mapStateToProps = (state, ownProps) => ({
  quantity: getProductQuantity(ownProps.productDetail.id)(state)
});

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({
    handleChange: props => (v) => {
      props.updateProduct({
        product: props.productDetail,
        quantity: v
      })
    }
  })
)

export default enhance(Product)

