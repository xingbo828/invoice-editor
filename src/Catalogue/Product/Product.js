import React from 'react';

import { Container, Image, Title, Price } from './styles';
import NumberField from './NumberField';

const Product = ({ handleChange, productDetail, quantity, unit }) => (
  <Container>
    <Image src={productDetail.image} />
    <Title>{productDetail.title}</Title>
    <Price>${productDetail.price}</Price>
    <NumberField onChange={handleChange} unit={unit} value={quantity} />
  </Container>
);

export default Product;
