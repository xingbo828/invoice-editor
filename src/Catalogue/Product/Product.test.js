import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';

const props = {
  handleChange: () => {},
  productDetail: {
    title: 'A',
    image: 'B',
    price: 25
  },
  quantity: 3,
  unit: 'Qty'
}
it('should match snapshot', () => {
  const wrapper = shallow(<Product {...props} />);
  expect(wrapper).toMatchSnapshot()
});