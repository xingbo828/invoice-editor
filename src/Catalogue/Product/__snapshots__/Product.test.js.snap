// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    handleChange={[Function]}
    productDetail={
      Object {
        "image": "B",
        "price": 25,
        "title": "A",
      }
    }
    quantity={3}
    unit="Qty"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          src="B"
        />,
        <ForwardRef>
          A
        </ForwardRef>,
        <ForwardRef>
          $
          25
        </ForwardRef>,
        <NumberField
          label=""
          onChange={[Function]}
          placeholder=""
          unit="Qty"
          value={3}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "B",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
  margin: 0 auto 1.5rem;
  width: 100%;
  height: 100px;
  background: url('",
              [Function],
              "') center center no-repeat;
  background-size: contain;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "A",
        },
        "ref": null,
        "rendered": "A",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "$",
            25,
          ],
        },
        "ref": null,
        "rendered": Array [
          "$",
          25,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "",
          "onChange": [Function],
          "placeholder": "",
          "unit": "Qty",
          "value": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  text-align: center;
  margin-bottom: 1rem;
  display: inline-block;
  padding: 2rem;
  box-shadow: ",
          [Function],
          ";
  border-radius: 4px;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.li",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "li",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            src="B"
          />,
          <ForwardRef>
            A
          </ForwardRef>,
          <ForwardRef>
            $
            25
          </ForwardRef>,
          <NumberField
            label=""
            onChange={[Function]}
            placeholder=""
            unit="Qty"
            value={3}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "B",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
  margin: 0 auto 1.5rem;
  width: 100%;
  height: 100px;
  background: url('",
                [Function],
                "') center center no-repeat;
  background-size: contain;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "A",
          },
          "ref": null,
          "rendered": "A",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "$",
              25,
            ],
          },
          "ref": null,
          "rendered": Array [
            "$",
            25,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
  padding: 0 0 1rem;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.5;
  color: ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "",
            "onChange": [Function],
            "placeholder": "",
            "unit": "Qty",
            "value": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  width: 100%;
  text-align: center;
  margin-bottom: 1rem;
  display: inline-block;
  padding: 2rem;
  box-shadow: ",
            [Function],
            ";
  border-radius: 4px;
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.li",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "li",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
